### PR TITLE
GSDX: Move PSX code to an isolated path

### PR DESCRIPTION
**Pull request by ssakash**
**Summary of Changes**:

* Move PSX code of GSDX to a separate directory, prevents confusion for newbies who are interested on contributing to the PSX part of the GSDX. Compiles fine on VS2015, untested on Linux.

**Edit**: Apparently the Linux build is failing (looking at you, Travis CI). any linux coders want to do the linux counterpart of the changes ?

### CODE REVIEW

**Comment by gregory38** [comment]
You need to update CmakeList.txt file to update the path too.

However, I don't like your change of all include file with `../` so I propose that you update the Include directory path in compilers

File: plugins/GSdx/GSdx.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #include "stdafx.h"
 #include "GSdx.h"
 #include "GS.h"
-#include "GPU.h"
+#include "../GSdx/PSX/GPU.h"
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSdx.cpp]: Why is /GSdx on there? you could have just had "./PSX/GPU.h"

File: plugins/GSdx/PSX/GPU.h [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 #pragma pack(push, 1)
 
-#include "GS.h"
+#include "../gs.h"
```
Comment (refractionpcsx2) [inline plugins/GSdx/PSX/GPU.h]: Linux could be anal about case here

Comment (gregory38) [inline plugins/GSdx/PSX/GPU.h]: s/could be/is/ ;)

**Comment by refractionpcsx2** [comment]
> However, I don't like your change of all include file with ../ so I propose that you update the Include directory path in compilers

Hmm I don't like that idea either, messing with include paths like that is messy with VS :P

**Comment by gregory38** [comment]
What is the issue with VS ?

**Comment by refractionpcsx2** [comment]
It's just a PITA because in VS you have multiple locations for includes >.<

**Comment by gregory38** [comment]
What do you mean? Same include name in multiple directory?

**Comment by refractionpcsx2** [comment]
In VS you have multiple places to include directories, one is in the "Project settings" and one is in the microsoft.cpp.user resource. There's just bits all over the place.

Anyways I don't know why we should do it specifically this way when we haven't done it like that anywhere else :)

**Comment by gregory38** [comment]
I'm not sure x86emitter/common use full path. Can't you just set the include path in GSdx vs files ?

I'm personally think it would be nicer to not bother with the dot stuff in path.

**Comment by gregory38** [comment]
See it is used on common directory
```
gr AdditionalIncludeDirectories common 
common/vsprops/common.props:      <AdditionalIncludeDirectories>"$(SolutionDir)3rdparty";"$(SolutionDir)3rdparty\w32pthreads\include";"$(SolutionDir)common\include";%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
common/vsprops/common.props:      <AdditionalIncludeDirectories>$(SolutionDir)common\include</AdditionalIncludeDirectories>
common/vsprops/BaseProperties.props:      <AdditionalIncludeDirectories>$(SvnRootDir)\common\include;$(SvnRootDir)\3rdparty\wxWidgets3.0\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
common/vsprops/BaseProperties.props:      <AdditionalIncludeDirectories>$(SolutionDir)\common\include;$(SolutionDir)\3rdparty\wxwidgets3.0\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
common/vsprops/CommonLibrary.props:      <AdditionalIncludeDirectories>$(SvnRootDir)\common\include\$(ProjectName);$(SvnRootDir)\common\include;$(SvnRootDir)\3rdparty\wxwidgets3.0\include;$(SvnRootDir)\3rdparty;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
common/vsprops/pthreads.props:      <AdditionalIncludeDirectories>$(SolutionDir)3rdparty\pthreads4w\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
common/vsprops/plugin_svnroot.props:      <AdditionalIncludeDirectories>$(ProjectDir);$(ProjectDir)\..\3rdparty;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
common/vsprops/3rdpartyDeps.props:      <AdditionalIncludeDirectories>$(SvnRootDir)\3rdparty\;$(SvnRootDir)\3rdparty\soundtouch\;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```

Actually even in GSdx. I don't think it will hurt to add `$(SolutionDir)/plugins/GSdx` in the middle.
```
gr AdditionalIncludeDirectories plugins/GSdx
plugins/GSdx/vsprops/common.props:      <AdditionalIncludeDirectories>$(VTUNE_AMPLIFIER_XE_2015_DIR)include;$(SolutionDir)unfree;$(SolutionDir)3rdparty;$(SolutionDir)3rdparty\libpng;$(SolutionDir)3rdparty\opencl;$(SolutionDir)3rdparty\zlib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```

**Comment by gregory38** [comment]
@refractionpcsx2 still no?
@turtleli any opinion ?

File: plugins/GSdx/vsprops/common.props [hunk 3]
```diff
@@ -13,7 +13,7 @@
       <WarningLevel>Level4</WarningLevel>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
       <DisableSpecificWarnings>4456;4458;4996;4995;4324;4100;4101;4201;4556;4127;4512;%(DisableSpecificWarnings)</DisableSpecificWarnings>
-      <AdditionalIncludeDirectories>$(VTUNE_AMPLIFIER_XE_2015_DIR)include;$(SolutionDir)unfree;$(SolutionDir)3rdparty;$(SolutionDir)3rdparty\libpng;$(SolutionDir)3rdparty\opencl;$(SolutionDir)3rdparty\zlib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(VTUNE_AMPLIFIER_XE_2015_DIR)include;$(SolutionDir)unfree;$(SolutionDir)3rdparty;$(SolutionDir)3rdparty\libpng;$(SolutionDir)plugins\GSdx;$(SolutionDir)3rdparty\opencl;$(SolutionDir)3rdparty\zlib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (turtleli) [inline plugins/GSdx/vsprops/common.props]: Did you test if the additional include directories actually need to be changed?

**Comment by refractionpcsx2** [comment]
> @refractionpcsx2 still no?

well, he's done it now :P Guess I'm outnumbered.  Democracy!

File: plugins/GSdx/vsprops/common.props (hunk 3 above)
Comment (gregory38) [inline plugins/GSdx/vsprops/common.props]: Good question !

**Comment by gregory38** [comment]
Free project is about dictatorship ;) Seriously I'm really open to suggestion.

File: plugins/GSdx/vsprops/common.props (hunk 3 above)
Comment (ssakash) [inline plugins/GSdx/vsprops/common.props]: > Did you test if the additional include directories actually need to be changed?

Yeah, right. I simply added it without even testing whether it's necessary,

Comment (turtleli) [inline plugins/GSdx/vsprops/common.props]: Did you try `$(ProjectDir)`?

Comment (ssakash) [inline plugins/GSdx/vsprops/common.props]: Changed it to ``$(ProjectDir)``

File: plugins/GSdx/vsprops/common.props [hunk 4]
```diff
@@ -13,7 +13,7 @@
       <WarningLevel>Level4</WarningLevel>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
       <DisableSpecificWarnings>4456;4458;4996;4995;4324;4100;4101;4201;4556;4127;4512;%(DisableSpecificWarnings)</DisableSpecificWarnings>
-      <AdditionalIncludeDirectories>$(VTUNE_AMPLIFIER_XE_2015_DIR)include;$(SolutionDir)unfree;$(SolutionDir)3rdparty;$(SolutionDir)3rdparty\libpng;$(SolutionDir)3rdparty\opencl;$(SolutionDir)3rdparty\zlib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(VTUNE_AMPLIFIER_XE_2015_DIR)include;$(ProjectDir);$(SolutionDir)unfree;$(SolutionDir)3rdparty;$(SolutionDir)3rdparty\libpng;$(SolutionDir)3rdparty\opencl;$(SolutionDir)3rdparty\zlib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (gregory38) [inline plugins/GSdx/vsprops/common.props]: Is projectdir correct? Or is it a left over ?

Comment (turtleli) [inline plugins/GSdx/vsprops/common.props]: It's correct.
https://msdn.microsoft.com/en-us/library/c02as0cs.aspx

Comment (gregory38) [inline plugins/GSdx/vsprops/common.props]: Ok. It was to be sure.

**Pull request merged**